### PR TITLE
Add system.storage

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: lib/system/storage.toit [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+_client_/StorageServiceClient? ::= (StorageServiceClient).open
+    --if_absent=: null
+
+class Bucket extends ServiceResourceProxy:
+  static SCHEME_RAM   ::= "ram"
+  static SCHEME_FLASH ::= "flash"
+
+  constructor.internal_ client/StorageServiceClient handle/int:
+    super client handle
+
```
Comment (erikcorry) [inline lib/system/storage.toit:23]: Doc comments to explain the uri format, what a Bucket is etc.

File: lib/system/storage.toit [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    return open --scheme=SCHEME_RAM --path=path
+
+  static open --flash/bool path/string -> Bucket:
+    if flash != true: throw "Bad Argument"
+    return open --scheme=SCHEME_FLASH --path=path
+
+  static open --scheme/string --path/string -> Bucket:
+    client := _client_
+    if not client: throw "UNSUPPORTED"
+    if (path.index_of ":") >= 0: throw "Paths cannot contain ':'"
```
Comment (erikcorry) [inline lib/system/storage.toit:40]: Perhaps use the --if_absent version

File: lib/system/storage.toit [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    handle := client.open_bucket --scheme=scheme --path=path
+    return Bucket.internal_ client handle
+
+  get key/string:
```
Comment (erikcorry) [inline lib/system/storage.toit:44]: Returns Object? ?

**Review by erikcorry — APPROVED** [review]

File: lib/system/storage.toit (hunk 3 above)
Comment (kasperl) [reply lib/system/storage.toit:44]: That would require downcasts.

File: lib/system/storage.toit (hunk 2 above)
Comment (kasperl) [reply lib/system/storage.toit:40]: Done.

File: lib/system/storage.toit (hunk 1 above)
Comment (kasperl) [reply lib/system/storage.toit:23]: Done.

**Pull request merged**